src/parser.rs

```diff
@@ -734,15 +738,14 @@ impl<'iter, 'item, I: Iterator<Item = (usize, &'item [u8])>> FallibleIterator
                 })??;
 
             if key == self.current {
```

Comment:
You could also do

```suggestion
            return if key == self.current {
```

And add a `;` at the final `}`. But works for me either way :)